static/scripts/dashboard/dashboardNewCtrl.js

```diff
@@ -3677,14 +3684,27 @@
     }
 
     function updateProvisionSystemIntegration(next) {
-      var bag = {
-        name: 'provision',
-        masterName: $scope.vm.installForm.provision.amazonKeys.masterName,
-        data: $scope.vm.installForm.provision.amazonKeys.data,
-        isEnabled: $scope.vm.installForm.provision.amazonKeys.isEnabled
-      };
+      async.eachSeries($scope.vm.installForm.provision,
+        function (provisionIntegration, nextIntegration) {
+          var bag = {
+            name: 'provision',
+            masterName: provisionIntegration.masterName,
+            data: provisionIntegration.data,
+            isEnabled: provisionIntegration.isEnabled
+          };
 
-      updateSystemIntegration(bag,
+          // TODO: Remove this once all the references in the system
+          // have been updated to correctly handle multiple "provision"
+          // providers.
+          if (bag.masterName === 'gcloudKey')
+            bag.name = 'provision-gcloud';
```

Comment:
We have a few places in API which refer to provision integration without the `masterName` which will get confused when we have two such integrations, these references will get cleaned up along the way. Once we have cleaned it all up, this will be back to provision.